apps/modeler/src/scene_tree.cpp

```diff
@@ -265,12 +259,8 @@ pcl::modeler::SceneTree::closePointCloud(const QList<CloudMeshItem*>& items)
 {
   QList<RenderWindowItem*> render_window_items;
 
-  for (QList<CloudMeshItem*>::const_iterator items_it = items.begin();
-    items_it != items.end();
-    ++ items_it)
+  for (auto item : items)
```

Comment:
Reference

Reply:
`item` is just a pointer, so it's a primitive data type, where I don't have references ;)

Reply:
Well following your own logic, you shouldn't have not used `auto` then, because `auto` is only for complex types ;)

Seriously though, looking at the diff I can see that item is a pointer. But without the diff it's just `auto item` of whatever type. Then the question is why copy? Why not `const`? Do the functions that are called on this `item` modify the object? If so, is it intended that effects are lost because it was a copy? And so on and so forth.

I really don't get this notion of coming up with funky rules of what should be `auto`ed, what should be  `&`ed... And now we reviewers have to check with microscope every occasion and understand the situation. This could have been `const auto& item`. And no questions to ask anymore, because it's crystal clear that we are just "viewing" the items in a container.

Reply:
You are right, in case `auto` is used it is maybe better to use always `const auto&`, because the type isn't visible at first moment.